src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php

```diff
@@ -11,36 +11,42 @@
 
 namespace Symfony\Component\Form\Extension\Core\Type;
 
-use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Intl\Intl;
-use Symfony\Component\OptionsResolver\OptionsResolver;
 
-class CurrencyType extends AbstractType
+class CurrencyType extends ImmutableChoiceType
 {
     /**
-     * {@inheritdoc}
+     * Stores the available currency choices.
      */
-    public function configureOptions(OptionsResolver $resolver)
-    {
-        $resolver->setDefaults(array(
-            'choices' => array_flip(Intl::getCurrencyBundle()->getCurrencyNames()),
-            'choice_translation_domain' => false,
-        ));
-    }
+    private static $currencies;
 
     /**
      * {@inheritdoc}
      */
-    public function getParent()
+    public function getBlockPrefix()
     {
-        return __NAMESPACE__.'\ChoiceType';
+        return 'currency';
     }
 
     /**
      * {@inheritdoc}
+     *
+     * Returns the currency choices.
+     *
+     * The choices are generated from the Intl component
+     * \Symfony\Component\Intl\Intl::getCurrencyBundle().
+     * They are cached during a single request, so multiple
+     * currency fields on the same page don't lead to
+     * unnecessary overhead.
+     *
+     * @return array The currency choices
      */
-    public function getBlockPrefix()
+    protected function loadChoices()
     {
-        return 'currency';
+        if (null === self::$currencies) {
```

Comment:
storing them in a static property will increase the memory usage by preventing garbage collection until the end of the process.
What kind of performance gain is there in term of time here, to counter-balance the loss regarding memory perf ?